lib/peep.rb

```diff
@@ -0,0 +1,85 @@
+require_relative 'database_connection'
+
+class Peep
+
+  def self.all
+
+    sql = %{select id, peep, parent_peep, posted_datetime, user_id
+      from peeps order by id desc}
+    peeps = DatabaseConnection.query(sql)
+    peeps.map { |record| { id: record["id"],
+      peep: record["peep"],
+      posted_date: Peep.date_only(record["posted_datetime"]),
+      peeper: find_username(record["user_id"]),
+      parent_peep: record["parent_peep"],
+      peeper_id: record["user_id"] }
+    }
+  end
+
+  def self.create(message, user_id, parent_peep = 0)
+
+    sql = %{INSERT INTO peeps
+      (peep, user_id, parent_peep) VALUES
+      ('#{message}', '#{user_id}', '#{parent_peep}')
+      RETURNING id, peep, parent_peep, posted_datetime;}
+    newpeep = DatabaseConnection.query(sql)
+    id = newpeep[0]["id"]
+
+    Peep.new(id, message, user_id, parent_peep)
+  end
+
+  def self.reply(message, user_id, parent_peep)
+    create(message, user_id, parent_peep)
+
+  end
+
+  def self.date_only(date)
```

Comment:
Could this be a private method?  Making instance methods private in Ruby requires a bit of effort but it is possible